### PR TITLE
ARACHNE-2370 Add filter by tenant into Admin Panel

### PR DESCRIPTION
**Pull request by sergey-maletsky**

### CODE REVIEW

File: src/main/java/com/odysseusinc/arachne/portal/model/search/UserSearch.java [hunk 1]
```diff
@@ -56,4 +57,14 @@ public void setQuery(String query) {
 
         this.query = query;
     }
+
+    public Long[] getTenant() {
+
```
Comment (wivern) [inline src/main/java/com/odysseusinc/arachne/portal/model/search/UserSearch.java]: I guess it should be named like 'getTenantIds'

File: src/main/java/com/odysseusinc/arachne/portal/repository/UserSpecifications.java [hunk 2]
```diff
@@ -48,6 +54,19 @@
         return (root, query, cb) -> cb.like(root.get(field), namePattern);
     }
 
+    public static <U extends IUser> Specification<U> usersIn(final Long[] tenantIds, EntityManager em) {
+
+        return ((root, criteriaQuery, cb) -> {
+            List<Long> tenantList = java.util.Arrays.asList(tenantIds);
+            Query query = em.createNativeQuery(USERS_DATA_JOIN_TENANTS_USERS_IN_QUERY);
+            query.setParameter("tenants", tenantList);
+            query.setParameter("count", tenantList.size() - 1);
```
Comment (wivern) [inline src/main/java/com/odysseusinc/arachne/portal/repository/UserSpecifications.java]: This is not clear to me, could you explain some more details?

File: src/main/java/com/odysseusinc/arachne/portal/api/v1/controller/BaseAdminController.java [hunk 3]
```diff
@@ -147,19 +147,10 @@ public void fullDeleteStudy(Long[] ids) {
...
-        Iterator<Sort.Order> pageIt = pageable.getSort().iterator();
-        Stream<Sort.Order> pageStream = StreamSupport.stream(Spliterators.spliteratorUnknownSize(pageIt, Spliterator.ORDERED), false);
-        if (pageStream.anyMatch(order -> order.getProperty().equals("name"))) {
-            search = new PageRequest(pageable.getPageNumber() - 1, pageable.getPageSize(),
-                    pageable.getSort().getOrderFor("name").getDirection(),
-                    "firstname", "middlename", "lastname");
-        }
-        Page<IUser> users = userService.getAll(search, userSearch);
-        return users
-                .map(user -> conversionService.convert(user, CommonUserDTO.class));
```
Comment (agackovka) [inline src/main/java/com/odysseusinc/arachne/portal/api/v1/controller/BaseAdminController.java]: Unused imports?)

File: src/main/java/com/odysseusinc/arachne/portal/repository/UserSpecifications.java [hunk 4]
```diff
@@ -48,6 +50,17 @@
         return (root, query, cb) -> cb.like(root.get(field), namePattern);
     }
 
+    public static <U extends IUser> Specification<U> usersIn(final Long[] tenantIds) {
+
+        return ((root, query, cb) -> {
+            final Path<Tenant> tenantIdPath = root.join("tenants").get("id");
+            query.groupBy(root.get("id"));
+            query.having(cb.gt(cb.count(root), tenantIds.length - 1));
```
Comment (wivern) [inline src/main/java/com/odysseusinc/arachne/portal/repository/UserSpecifications.java]: Could you explain me how length of identities array related with number of users within tenant?

File: src/main/java/com/odysseusinc/arachne/portal/service/impl/BaseUserServiceImpl.java [hunk 5]
```diff
@@ -619,7 +626,26 @@ public U getByUuidAndInitializeCollections(String uuid) {
...
+        final Long[] tenantIds = userSearch.getTenantIds();
+        if (tenantIds != null && tenantIds.length > 0) {
+            spec = spec.and(usersIn(tenantIds));
+        }
+
+        return rawUserRepository.findAll(spec, search);
+    }
+
+    private Pageable convertOrderRequest(Pageable pageable) {
+        Pageable search = new PageRequest(pageable.getPageNumber() - 1, pageable.getPageSize(), pageable.getSort());
```
Comment (wivern) [inline src/main/java/com/odysseusinc/arachne/portal/service/impl/BaseUserServiceImpl.java]: why not to move this into if-else below?

**Pull request merged**